### PR TITLE
Migrate vue to vite.

### PR DESCRIPTION
**Pull request by mshima**
- rework i18n, build and load using vite instead of using axios request.
- microfrontends is kept using webpack, vite-module-federation is not working as expected.

Related to https://github.com/jhipster/generator-jhipster/issues/23416.

---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/vue/templates/package.json.ejs [hunk 1]
```diff
@@ -96,26 +114,17 @@
     "cypress": "<%= nodeDependencies['cypress'] %>",
 <%_ } _%>
     "rimraf": "<%= nodeDependencies['rimraf'] %>",
+    "rollup-plugin-copy": null,
```
Comment (qmonmert) [inline generators/vue/templates/package.json.ejs:117]: it's mandatory?

File: generators/vue/templates/package.json.ejs [hunk 2]
```diff
@@ -96,26 +114,17 @@
...
-    "sass-loader": "<%= nodeDependencies['sass-loader'] %>",
     "sinon": "<%= nodeDependencies['sinon'] %>",
     "swagger-ui-dist": "<%= nodeDependencies['swagger-ui-dist'] %>",
-    "terser-webpack-plugin": "<%= nodeDependencies['terser-webpack-plugin'] %>",
-    "ts-loader": "<%= nodeDependencies['ts-loader'] %>",
     "eslint": "<%= nodeDependencies['eslint'] %>",
     "eslint-plugin-prettier": "<%= nodeDependencies['eslint-plugin-prettier'] %>",
     "eslint-plugin-vue": "<%= nodeDependencies['eslint-plugin-vue'] %>",
     "typescript": "<%= nodeDependencies['typescript'] %>",
+    "vite": null,
```
Comment (qmonmert) [inline generators/vue/templates/package.json.ejs:125]: same?

File: generators/vue/templates/tsconfig.vitest.json.ejs [hunk 3]
```diff
@@ -0,0 +1,9 @@
+{
+  "extends": "./tsconfig.app.json",
+  "exclude": [],
```
Comment (qmonmert) [inline generators/vue/templates/tsconfig.vitest.json.ejs:3]: we can delete this line?

File: generators/vue/templates/tsconfig.vitest.json.ejs [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+  "compilerOptions": {
+    "composite": true,
+    "lib": [],
```
Comment (qmonmert) [inline generators/vue/templates/tsconfig.vitest.json.ejs:6]: same?

File: generators/vue/templates/package.json.ejs (hunk 1 above)
Comment (mshima) [reply generators/vue/templates/package.json.ejs:117]: I’ve added a transform that replaces null values with the value from nodeDependencies.
Maybe this is too confusing?

File: generators/vue/templates/tsconfig.vitest.json.ejs (hunk 3 above)
Comment (mshima) [reply generators/vue/templates/tsconfig.vitest.json.ejs:3]: This exclude overrides exclude from the extended configuration file.

Comment (qmonmert) [reply generators/vue/templates/tsconfig.vitest.json.ejs:3]: ok

File: generators/vue/templates/package.json.ejs (hunk 1 above)
Comment (qmonmert) [reply generators/vue/templates/package.json.ejs:117]: ok a little, but ok

**Review by qmonmert — APPROVED** [review]

File: generators/vue/templates/tsconfig.vitest.json.ejs (hunk 4 above)
Comment (mshima) [reply generators/vue/templates/tsconfig.vitest.json.ejs:6]: Not sure, this is generated by `npm init create-vue`.
It probably overrides the lib from extended config.

**Pull request merged**